src/text_buffer.rs

```diff
@@ -532,11 +544,23 @@ impl TextBuffer {
 
     pub fn undo(&mut self) -> bool {
         let state = self.history.undo(&mut self.row);
+        if let Some((_, _, _, ongoing)) = state {
+            if !ongoing {
```

Comment:
Isn't this condition opposite? The bool value represents some diff appears or not derived from `finish_ongoing_edit()`. It means that modification occurred or not. So `true` means text was modified and `self_modified_count` should be updated.

Reply:
If there were some changes not put into history (`edited = true`),
these changes are canceled by undo,
so modified count keeps as is.

In the other hand, if `edited = false`,
then undo reverts last history
so modified count should be decremented.

Reply:
oh, ok, I got the point. I misunderstood the logic.